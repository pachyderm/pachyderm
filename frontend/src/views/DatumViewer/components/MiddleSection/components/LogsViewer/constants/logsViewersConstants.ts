import {DropdownItem} from '@pachyderm/components';

export const DEFAULT_ROW_HEIGHT = 24;
export const HEADER_HEIGHT_OFFSET = 40;

export const LOGS_DEFAULT_DROPDOWN_OPTIONS: DropdownItem[] = [
  {id: 'Last 30 Minutes', content: 'Last 30 Minutes'},
  {id: 'Last 24 Hours', content: 'Last 24 Hours'},
  {id: 'Last 3 Days', content: 'Last 3 Days'},
];

export const LOGS_PAGE_SIZE = 1000;
