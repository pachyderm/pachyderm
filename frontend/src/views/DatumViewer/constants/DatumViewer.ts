//TODO: Update once we get regex
export const DATUM_ID_LENGTH = 64;
export const JOB_DATE_FORMAT = 'MMM dd, yyyy HH:mmaaa';
