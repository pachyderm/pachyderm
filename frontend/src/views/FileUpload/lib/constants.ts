export const GLOB_CHARACTERS = '* ? [ ] { } ! ( ) @ + ^';
export const ERROR_MESSAGE =
  'Sorry that we experienced some techincal difficulties, Please try uploading again.';
