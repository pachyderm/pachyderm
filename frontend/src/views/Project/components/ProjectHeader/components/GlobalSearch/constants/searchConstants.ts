export const MAX_HISTORY = 5;
