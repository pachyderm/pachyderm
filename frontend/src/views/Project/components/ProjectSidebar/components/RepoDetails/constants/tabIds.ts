export enum TAB_ID {
  INFO = 'info',
  COMMITS = 'commits',
}
