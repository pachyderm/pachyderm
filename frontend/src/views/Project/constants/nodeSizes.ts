export const NODE_WIDTH = 256;
export const NODE_HEIGHT = 128;
export const NODE_INPUT_REPO = 48;
