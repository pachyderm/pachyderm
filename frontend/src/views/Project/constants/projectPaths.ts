/* eslint-disable no-useless-escape */
/* eslint-disable prettier/prettier */
export const PROJECT_PATH = '/project/:projectId';
export const PROJECT_REPOS_PATH = '/project/:projectId/repos';
export const PROJECT_PIPELINES_PATH = '/project/:projectId/pipelines';

export const PROJECT_REPO_PATH = '/project/:projectId/repos/:repoId/branch/:branchId/:tabId?';
export const PROJECT_PIPELINE_PATH = '/project/:projectId/pipelines/:pipelineId/:tabId?';
export const PROJECT_JOBS_PATH = '/project/:projectId/jobs';
export const PROJECT_JOB_PATH = '/project/:projectId/jobs/:jobId/:pipelineId?';
export const PROJECT_PIPELINE_JOB_PATH = '/project/:projectId/jobs/:jobId/:pipelineId';
export const PROJECT_FILE_UPLOAD_PATH = '/project/:projectId/repos/:repoId/upload';

export const PROJECT_FILE_BROWSER_PATH = '/project/:projectId/repos/:repoId/branch/:branchId/commit/:commitId/:filePath?';

export const PROJECT_LOGS_VIEWER_PIPELINE_PATH = `/project/:projectId/pipelines/:pipelineId/logs`;
export const PROJECT_LOGS_VIEWER_JOB_PATH = `/project/:projectId/jobs/:jobId/:pipelineId/logs`;

export const LINEAGE_PATH = '/lineage/:projectId'

export const LINEAGE_REPOS_PATH = '/lineage/:projectId/repos'
export const LINEAGE_PIPELINES_PATH = '/lineage/:projectId/pipelines'

export const LINEAGE_REPO_PATH = '/lineage/:projectId/repos/:repoId/branch/:branchId/:tabId?';;
export const LINEAGE_PIPELINE_PATH = '/lineage/:projectId/pipelines/:pipelineId/:tabId?';
export const LINEAGE_JOBS_PATH = '/lineage/:projectId/jobs';
export const LINEAGE_JOB_PATH = '/lineage/:projectId/jobs/:jobId/:pipelineId?';
export const LINEAGE_PIPELINE_JOB_PATH = '/lineage/:projectId/jobs/:jobId/:pipelineId';
export const LINEAGE_FILE_UPLOAD_PATH = '/lineage/:projectId/repos/:repoId/upload';

export const LINEAGE_FILE_BROWSER_PATH = '/lineage/:projectId/repos/:repoId/branch/:branchId/commit/:commitId/:filePath?';

export const LINEAGE_LOGS_VIEWER_PIPELINE_PATH = `/lineage/:projectId/pipelines/:pipelineId/logs`;
export const LINEAGE_LOGS_VIEWER_JOB_PATH = `/lineage/:projectId/jobs/:jobId/:pipelineId/logs`;

export const PROJECT_PATHS = [
  LINEAGE_PATH,
  LINEAGE_REPOS_PATH,
  LINEAGE_PIPELINES_PATH,
  LINEAGE_JOB_PATH,
  LINEAGE_JOBS_PATH,
  LINEAGE_PIPELINE_JOB_PATH,
  LINEAGE_REPO_PATH,
  LINEAGE_PIPELINE_PATH,
  LINEAGE_FILE_UPLOAD_PATH,
  LINEAGE_FILE_BROWSER_PATH,
  LINEAGE_LOGS_VIEWER_PIPELINE_PATH,
  LINEAGE_LOGS_VIEWER_JOB_PATH,
  PROJECT_PATH,
  PROJECT_REPOS_PATH,
  PROJECT_PIPELINES_PATH,
  PROJECT_JOBS_PATH,
  PROJECT_JOB_PATH,
  PROJECT_PIPELINE_JOB_PATH,
  PROJECT_REPO_PATH,
  PROJECT_PIPELINE_PATH,
  PROJECT_FILE_UPLOAD_PATH,
  PROJECT_FILE_BROWSER_PATH,
  PROJECT_LOGS_VIEWER_PIPELINE_PATH,
  PROJECT_LOGS_VIEWER_JOB_PATH,
];
