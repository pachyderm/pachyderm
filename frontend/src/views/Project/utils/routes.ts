import {ExtractRouteParams, generatePath, matchPath} from 'react-router';

import {generatePathWithSearch} from '@pachyderm/components';

import {
  LINEAGE_PATH,
  PROJECT_JOBS_PATH,
  PROJECT_PIPELINE_PATH,
  PROJECT_PIPELINE_JOB_PATH,
  PROJECT_PATH,
  PROJECT_REPOS_PATH,
  PROJECT_PIPELINES_PATH,
  PROJECT_REPO_PATH,
  LINEAGE_FILE_BROWSER_PATH,
  PROJECT_FILE_BROWSER_PATH,
  PROJECT_JOB_PATH,
  LINEAGE_LOGS_VIEWER_JOB_PATH,
  LINEAGE_LOGS_VIEWER_PIPELINE_PATH,
  PROJECT_LOGS_VIEWER_JOB_PATH,
  PROJECT_LOGS_VIEWER_PIPELINE_PATH,
  LINEAGE_JOBS_PATH,
  LINEAGE_JOB_PATH,
  LINEAGE_PIPELINE_JOB_PATH,
  LINEAGE_REPO_PATH,
  LINEAGE_PIPELINE_PATH,
  PROJECT_FILE_UPLOAD_PATH,
  LINEAGE_FILE_UPLOAD_PATH,
} from '../constants/projectPaths';

const generateRouteFn = <S extends string>(path: S) => {
  return (params?: ExtractRouteParams<S>, withSearch = true) => {
    return withSearch
      ? generatePathWithSearch(path, params)
      : encodeURI(generatePath(path, params));
  };
};

const generateLineageOrProjectRouteFn = <S extends string>(
  projectPath: S,
  lineagePath: S,
) => {
  return (params?: ExtractRouteParams<S>, withSearch = true) => {
    if (matchPath(window.location.pathname, LINEAGE_PATH)) {
      return withSearch
        ? generatePathWithSearch(lineagePath, params)
        : encodeURI(generatePath(lineagePath, params));
    } else {
      return withSearch
        ? generatePathWithSearch(projectPath, params)
        : encodeURI(generatePath(projectPath, params));
    }
  };
};

export const lineageRoute = generateRouteFn(LINEAGE_PATH);
export const projectRoute = generateRouteFn(PROJECT_PATH);
export const projectReposRoute = generateRouteFn(PROJECT_REPOS_PATH);
export const projectPipelinesRoute = generateRouteFn(PROJECT_PIPELINES_PATH);

export const jobRoute = generateLineageOrProjectRouteFn(
  PROJECT_JOB_PATH,
  LINEAGE_JOB_PATH,
);
export const pipelineJobRoute = generateLineageOrProjectRouteFn(
  PROJECT_PIPELINE_JOB_PATH,
  LINEAGE_PIPELINE_JOB_PATH,
);
export const jobsRoute = generateLineageOrProjectRouteFn(
  PROJECT_JOBS_PATH,
  LINEAGE_JOBS_PATH,
);
export const repoRoute = generateLineageOrProjectRouteFn(
  PROJECT_REPO_PATH,
  LINEAGE_REPO_PATH,
);
export const pipelineRoute = generateLineageOrProjectRouteFn(
  PROJECT_PIPELINE_PATH,
  LINEAGE_PIPELINE_PATH,
);

export const fileBrowserRoute = generateLineageOrProjectRouteFn(
  PROJECT_FILE_BROWSER_PATH,
  LINEAGE_FILE_BROWSER_PATH,
);
export const logsViewerJobRoute = generateLineageOrProjectRouteFn(
  PROJECT_LOGS_VIEWER_JOB_PATH,
  LINEAGE_LOGS_VIEWER_JOB_PATH,
);
export const logsViewerPipelneRoute = generateLineageOrProjectRouteFn(
  PROJECT_LOGS_VIEWER_PIPELINE_PATH,
  LINEAGE_LOGS_VIEWER_PIPELINE_PATH,
);

export const fileUploadRoute = generateLineageOrProjectRouteFn(
  PROJECT_FILE_UPLOAD_PATH,
  LINEAGE_FILE_UPLOAD_PATH,
);
