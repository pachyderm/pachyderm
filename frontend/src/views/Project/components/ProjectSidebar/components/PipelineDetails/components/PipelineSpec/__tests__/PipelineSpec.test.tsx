import {
  render,
  waitForElementToBeRemoved,
  screen,
  fireEvent,
  waitFor,
} from '@testing-library/react';
import {setupServer} from 'msw/lib/node';
import React from 'react';
import {Route} from 'react-router';

import {
  mockEmptyGetAuthorize,
  mockGetMontagePipeline,
} from '@dash-frontend/mocks';
import {withContextProviders} from '@dash-frontend/testHelpers';
import {LINEAGE_PIPELINE_PATH} from '@dash-frontend/views/Project/constants/projectPaths';
import {pipelineRoute} from '@dash-frontend/views/Project/utils/routes';

import PipelineSpecComponent from '..';

describe('PipelineSpec', () => {
  const server = setupServer();

  const PipelineSpec = withContextProviders(() => (
    <Route path={LINEAGE_PIPELINE_PATH} component={PipelineSpecComponent} />
  ));

  beforeAll(() => {
    server.listen();
    server.use(mockEmptyGetAuthorize());
    server.use(mockGetMontagePipeline());
  });

  afterAll(() => server.close());

  it('should correctly render JSON spec', async () => {
    window.history.replaceState(
      '',
      '',
      pipelineRoute({
        projectId: 'default',
        pipelineId: 'montage',
      }),
    );

    render(<PipelineSpec />);

    await waitForElementToBeRemoved(() => screen.queryByRole('status'));

    const codeSpec = await screen.findByTestId(
      'ConfigFilePreview__codeElement',
    );

    await waitFor(() => {
      expect(document.querySelectorAll('.cm-cursor-primary')).toHaveLength(1);
    }); // wait for cursor to appear

    expect(codeSpec).toMatchSnapshot();
  });
});
