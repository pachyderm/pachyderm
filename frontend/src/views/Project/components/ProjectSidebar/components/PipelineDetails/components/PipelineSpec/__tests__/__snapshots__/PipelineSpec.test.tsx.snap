// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PipelineSpec should correctly render JSON spec 1`] = `
<div
  class="base"
>
  <div>
    <div
      class="header"
    >
      <h5 />
      <div
        class="settings base horizontal spacing16"
      >
        <div
          class="base"
        >
          <button
            aria-expanded="false"
            aria-haspopup="true"
            class="base base dropdown"
            data-testid="DropdownButton__button"
            type="button"
          >
            <div
              class="base horizontal justifyCenter alignCenter spacing8"
            >
              <span
                class="children"
              >
                <div
                  class="base plum"
                >
                  <svg-mock />
                </div>
              </span>
            </div>
          </button>
          <div
            aria-hidden="true"
            class="base right"
            role="menu"
          >
            <button
              class="base"
              data-testid="DropdownMenuItem__button"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              <div
                class="base horizontal alignCenter"
              >
                <div>
                  Copy
                </div>
              </div>
            </button>
            <button
              class="base"
              data-testid="DropdownMenuItem__button"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              <div
                class="base horizontal alignCenter"
              >
                <div>
                  Download JSON
                </div>
              </div>
            </button>
            <button
              class="base"
              data-testid="DropdownMenuItem__button"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              <div
                class="base horizontal alignCenter"
              >
                <div>
                  Download YAML
                </div>
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="codeBody"
      data-testid="ConfigFilePreview__codeElement"
    >
      <div
        class="codeSection"
      >
        <div>
          <div
            class="code language-yaml depth0 hljs"
          />
          <div
            class="header"
          >
            <div
              class="base chevron black small"
            >
              <svg-mock />
            </div>
            <div
              class="code language-yaml depth0 hljs"
            >
              <span
                class="hljs-attr"
              >
                transform:
              </span>
            </div>
          </div>
          <div
            class=""
          >
            <div
              class="codeSection"
            >
              <div>
                <div
                  class="code language-yaml depth1 hljs"
                >
                  <span
                    class="hljs-attr"
                  >
                    image:
                  </span>
                   
                  <span
                    class="hljs-string"
                  >
                    v4tech/imagemagick
                  </span>
                  

                </div>
                <div
                  class="header"
                >
                  <div
                    class="base chevron black small"
                  >
                    <svg-mock />
                  </div>
                  <div
                    class="code language-yaml depth1 hljs"
                  >
                    <span
                      class="hljs-attr"
                    >
                      cmd:
                    </span>
                  </div>
                </div>
                <div
                  class=""
                >
                  <div
                    class="codeSection"
                  >
                    <div
                      class="code language-yaml depth2 hljs"
                    >
                      <span
                        class="hljs-bullet"
                      >
                        -
                      </span>
                       
                      <span
                        class="hljs-string"
                      >
                        sh
                      </span>
                      

                    </div>
                  </div>
                </div>
              </div>
              <div>
                <div
                  class="code language-yaml depth1 hljs"
                />
                <div
                  class="header"
                >
                  <div
                    class="base chevron black small"
                  >
                    <svg-mock />
                  </div>
                  <div
                    class="code language-yaml depth1 hljs"
                  >
                    <span
                      class="hljs-attr"
                    >
                      stdin:
                    </span>
                  </div>
                </div>
                <div
                  class=""
                >
                  <div
                    class="codeSection"
                  >
                    <div
                      class="code language-yaml depth2 hljs"
                    >
                      <span
                        class="hljs-bullet"
                      >
                        -
                      </span>
                       
                      <span
                        class="hljs-string"
                      >
                        &gt;-
  montage -shadow -background SkyBlue -geometry 300x300+2+2 $(find /pfs -type f
  | sort) /pfs/out/montage.png

                      </span>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="code language-yaml depth1 hljs"
              />
            </div>
          </div>
        </div>
        <div>
          <div
            class="code language-yaml depth0 hljs"
          />
          <div
            class="header"
          >
            <div
              class="base chevron black small"
            >
              <svg-mock />
            </div>
            <div
              class="code language-yaml depth0 hljs"
            >
              <span
                class="hljs-attr"
              >
                parallelismSpec:
              </span>
            </div>
          </div>
          <div
            class=""
          >
            <div
              class="codeSection"
            >
              <div
                class="code language-yaml depth1 hljs"
              >
                <span
                  class="hljs-attr"
                >
                  constant:
                </span>
                 
                <span
                  class="hljs-number"
                >
                  8
                </span>
                

              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="code language-yaml depth0 hljs"
          />
          <div
            class="header"
          >
            <div
              class="base chevron black small"
            >
              <svg-mock />
            </div>
            <div
              class="code language-yaml depth0 hljs"
            >
              <span
                class="hljs-attr"
              >
                input:
              </span>
            </div>
          </div>
          <div
            class=""
          >
            <div
              class="codeSection"
            >
              <div>
                <div
                  class="code language-yaml depth1 hljs"
                />
                <div
                  class="header"
                >
                  <div
                    class="base chevron black small"
                  >
                    <svg-mock />
                  </div>
                  <div
                    class="code language-yaml depth1 hljs"
                  >
                    <span
                      class="hljs-attr"
                    >
                      cross:
                    </span>
                  </div>
                </div>
                <div
                  class=""
                >
                  <div
                    class="codeSection"
                  >
                    <div
                      class="code language-yaml depth2 hljs"
                    >
                      <span
                        class="hljs-bullet"
                      >
                        -
                      </span>
                       
                      <span
                        class="hljs-attr"
                      >
                        pfs:
                      </span>
                      
    
                      <span
                        class="hljs-attr"
                      >
                        project:
                      </span>
                       
                      <span
                        class="hljs-string"
                      >
                        Solar-Panel-Data-Sorting
                      </span>
                      
    
                      <span
                        class="hljs-attr"
                      >
                        repo:
                      </span>
                       
                      <span
                        class="hljs-string"
                      >
                        edges
                      </span>
                      

                      <span
                        class="hljs-bullet"
                      >
                        -
                      </span>
                       
                      <span
                        class="hljs-attr"
                      >
                        pfs:
                      </span>
                      
    
                      <span
                        class="hljs-attr"
                      >
                        project:
                      </span>
                       
                      <span
                        class="hljs-string"
                      >
                        Solar-Panel-Data-Sorting
                      </span>
                      
    
                      <span
                        class="hljs-attr"
                      >
                        repo:
                      </span>
                       
                      <span
                        class="hljs-string"
                      >
                        images
                      </span>
                      

                    </div>
                  </div>
                </div>
              </div>
              <div
                class="code language-yaml depth1 hljs"
              />
            </div>
          </div>
        </div>
        <div>
          <div
            class="code language-yaml depth0 hljs"
          />
          <div
            class="header"
          >
            <div
              class="base chevron black small"
            >
              <svg-mock />
            </div>
            <div
              class="code language-yaml depth0 hljs"
            >
              <span
                class="hljs-attr"
              >
                schedulingSpec:
              </span>
            </div>
          </div>
          <div
            class=""
          >
            <div
              class="codeSection"
            >
              <div>
                <div
                  class="code language-yaml depth1 hljs"
                />
                <div
                  class="header"
                >
                  <div
                    class="base chevron black small"
                  >
                    <svg-mock />
                  </div>
                  <div
                    class="code language-yaml depth1 hljs"
                  >
                    <span
                      class="hljs-attr"
                    >
                      nodeSelector:
                    </span>
                  </div>
                </div>
                <div
                  class=""
                >
                  <div
                    class="codeSection"
                  >
                    <div
                      class="code language-yaml depth2 hljs"
                    >
                      <span
                        class="hljs-bullet"
                      >
                        -
                      </span>
                       
                      <span
                        class="hljs-bullet"
                      >
                        -
                      </span>
                       
                      <span
                        class="hljs-string"
                      >
                        disktype
                      </span>
                      
  
                      <span
                        class="hljs-bullet"
                      >
                        -
                      </span>
                       
                      <span
                        class="hljs-string"
                      >
                        ssd
                      </span>
                      

                    </div>
                  </div>
                </div>
              </div>
              <div
                class="code language-yaml depth1 hljs"
              >
                <span
                  class="hljs-attr"
                >
                  priorityClassName:
                </span>
                 
                <span
                  class="hljs-string"
                >
                  high-priority
                </span>
                

              </div>
            </div>
          </div>
        </div>
        <div
          class="code language-yaml depth0 hljs"
        />
      </div>
    </div>
  </div>
</div>
`;
